File: sop-java/src/main/java/sop/enums/SignAs.java
// SPDX-FileCopyrightText: 2021 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package sop.enums;

public enum SignAs {
    /**
     * Signature is made over the binary message.
     */
    Binary,

    /**
     * Signature is made over the message in text mode.
     */
    Text,
    ;

    @Override
    public String toString() {
        return super.toString().toLowerCase();
    }
}

File: sop-java/src/main/kotlin/sop/enums/SignAs.kt
// SPDX-FileCopyrightText: 2023 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package sop.enums

enum class SignAs(val mode: Int) {
    /** Signature is made over the binary message. */
    binary(0x00),
    /** Signature is made over the message in text mode. */
    text(0x01)
}
